### ISSUE TITLE
Attachments Should Support Link to an external cloud storage service

### ISSUE THREAD

**Issue by Viktova** [mailcow/mailcow-dockerized#1970]
**Is your feature request related to a problem? Please describe.**
When Forwarding emails having large attachments, it's pain in the ass to wait for the attachment to reupload the attached files to mailcow server again and then send it to the forwarded recipients.

**Describe the solution you'd like**
an option to link attachments to a custom cloud storage service like seafile or nextcloud (since both use the same way of handling files) - Seafile is way more recommended due to its performance when uploading files.

**Additional context**
Example: (both 1.com and 2.com are being handled by the same mailcow service/instance)

Sending email:
-----------------------------
To: 2@2.com
From 1@1.com
Subject: 11
Body: 1111
Attachment(s): 1.zip 1.png 1.db 1.exe
-----------------------------

2@2.com receives 1@1.com's email, and would like to forward it to 3@3.com (third party)
then he must wait for the attachments to reupload to mailcow (the same server been used by 1@1.com) (in case of 100+mb it would take like 2 mins or more to complete.

so instead of having these attachments downloaded they should be linked right after they been uploaded by 1@1.com the first time to a storage server / cloud server. and their converted to links. so that both 2@2.com and 3@3.com would download or attach / link them without waiting for the painful reupload.

**Comment by mkuron** [comment]
The trouble here is that this kind of replacement breaks DKIM signatures, so it needs to be done after rspamd validates the signature. If there is a Sieve rule that forwards the message elsewhere, the DKIM and ARC signatures also must still be intact so the forwarding destination doesn't reject the message. Unfortunately we can't place a hook after Sieve because that is already in the IMAP server's responsibility. We could only modify the message inside the IMAP mailbox, but that interferes with IMAP IDLE.

So in conclusion: this sounds like a useful feature, but it cannot be implemented server-side. You just need to ask 1@1.com to upload his files to a sharing service or configure his email client to do that automatically (I believe Thunderbird and Apple Mail can do that).

**Comment by mkuron** [comment]
Oh, I forgot you said that 1.com and 2.com are hosted on the same Mailcow server. It doesn't make a difference though: if 1@1.com signs or encrypts his messages with PGP or S/MIME, the signature would be broken. So replacing attachments with links definitely has to happen on the client.

**Comment by andryyy** [comment]
The only thing I could think of is to extract attachments and store them to be accessed in mailcow UI. Retention/size/file types could be defined as domain admin/admin/user. This does not make them available in a mail client though. You could forward them from within the UI maybe.
